aea/connections/base.py

```diff
@@ -40,17 +44,30 @@ def __init__(self):
 class Connection(ABC):
     """Abstract definition of a connection."""
 
-    def __init__(self, connection_id: str):
+    def __init__(self, connection_id: Optional[str] = None, supported_protocols: Optional[Set[str]] = None):
```

Comment:
now the base class `Connection` has an optional parameter `supported_protocols`.
That means both at the code level and configuration level the user can change the behaviour of the outgoing messages by ignoring the envelopes belonging to some protocols.

Reply:
also, worth noticing that this adds a burden to the developer of a new connection.

Reply:
See comments below. Valid points and I think I have an answer. Re making `connection_id` optional I am not sure. I could see this causing more confusion than being helpful.